p2p/pkg/rpc/validator/service.go

```diff
@@ -29,21 +32,41 @@ type Service struct {
 	logger          *slog.Logger
 	metrics         *metrics
 	optsGetter      func() (*bind.CallOpts, error)
+	notifier        notifications.Notifier
+	genesisTime     time.Time
 }
 
+var (
+	SlotDuration  = 12 * time.Second
+	EpochSlots    = 32
+	EpochDuration = SlotDuration * time.Duration(EpochSlots)
+	NotifyOffset  = 1 * time.Second
+	FetchOffset   = 2 * time.Second
+)
+
 func NewService(
 	apiURL string,
 	validatorRouter ValidatorRouterContract,
 	logger *slog.Logger,
 	optsGetter func() (*bind.CallOpts, error),
-) *Service {
-	return &Service{
+	notifier notifications.Notifier,
+) (*Service, error) {
+	s := &Service{
 		apiURL:          apiURL,
 		validatorRouter: validatorRouter,
 		logger:          logger,
 		metrics:         newMetrics(),
 		optsGetter:      optsGetter,
+		notifier:        notifier,
+	}
+	genesisTime, err := s.fetchGenesisTime(context.Background())
```

Comment:
The context should be passed in the `NewService` constructor or at least have a timeout.